### DISCUSSION TITLE
API Backend is not reading the .env file

### DISCUSSION THREAD

**Discussion by naman1512** [vercel/next.js#62271] [Help]
### Summary

I had made an app to design tshirts. I have integrated openAI API to generate images and embed them on the tshirt but I am getting the same errors and I am unable to fetch the data. 
<img width="971" alt="er2" src="https://github.com/vercel/next.js/assets/141447023/5321bfaa-9d1f-45ea-92d5-ec53951f8bc8">
<img width="1039" alt="er1" src="https://github.com/vercel/next.js/assets/141447023/7cdfffa2-9a36-40fd-b000-f4e8f0e032cc">

 OpenAIError: The OPENAI_API_KEY environment variable is missing or empty; either provide it, or instantiate the OpenAI client with an apiKey option, like new OpenAI({ apiKey: 'My API Key' }).

the following error is shown, I have shared my code. Please Help ME

**Comment by naman1512** [comment]
![image_2024-02-20_153539480](https://github.com/vercel/next.js/assets/141447023/318a0f1e-b56b-4c46-b22d-2e1ed1a15479)

Following is my Server Directory for backend
![image_2024-02-20_153649500](https://github.com/vercel/next.js/assets/141447023/db84bc1e-0eec-4862-a883-11844cfef167)
this is my index file.

Please Help

**Reply by icyJoseph** [reply]
could you move the `.env` file to the root of the project? outside of server. If that works, then you could still keep it within server by passing a path to `dotenv.config()` like:

```js
require('dotenv').config({ path: '/custom/path/to/.env' })
```
